perfkitbenchmarker/linux_benchmarks/dpb_distcp_benchmark.py

```diff
@@ -0,0 +1,159 @@
+# Copyright 2017 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Perform distributed copy of data on data processing backends.
+Apache Hadoop MapReduce distcp is an open-source tool used to copy large
+amounts of data. DistCp is very efficient because it uses MapReduce to copy the
+files or datasets and this means the copy operation is distributed across
+multiple nodes in a cluster.
+Benchmark to compare the performance of of the same distcp workload on clusters
+of various cloud providers.
+"""
+
+import copy
+import datetime
+
+
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import dpb_service
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+from perfkitbenchmarker.dpb_service import BaseDpbService
+
+BENCHMARK_NAME = 'dpb_distcp_benchmark'
+
+BENCHMARK_CONFIG = """
+dpb_distcp_benchmark:
+  description: Run distcp on dataproc and emr
+  dpb_service:
+    service_type: dataproc
+    worker_group:
+      vm_spec:
+        GCP:
+          machine_type: n1-standard-4
+          boot_disk_size: 1500
+        AWS:
+          machine_type: m3.xlarge
```

Comment:
m4s can only be run in a subnet, which can have implications on the performance. So the default implementation uses m3s. In either case, the actual machine types and worker counts will be overridden by cluster shapes and sizes closer to production.